cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -204,10 +208,15 @@ public ProgramController run(Program program, ProgramOptions options) {
       }
 
       boolean isLocal = SparkRuntimeContextConfig.isLocal(options);
-      SparkSubmitter submitter = isLocal
-        ? new LocalSparkSubmitter()
-        : new DistributedSparkSubmitter(hConf, locationFactory, host, runtimeContext,
-                                        options.getArguments().getOption(Constants.AppFabric.APP_SCHEDULER_QUEUE));
+      SparkSubmitter submitter;
+      if (sparkConfig != null) {
```

Comment:
guice injection isn't really meant for passing information around like this, but more for injecting the right implementation of an interface. It would be better to get the MasterEnvironment here, check whether it's null, and get the SparkConfig by directly calling it here.

Reply:
sounds good.